packages/web-config-server/src/tests/apiV1/dataBuilders/helpers/groupEvents.test.js

```diff
@@ -48,139 +30,353 @@ const PARENT_ORG_UNIT = {
+        }),
+      ).to.deep.equal([]));
+
+    it('finds all data element codes used in conditionals', () =>
+      expect(
+        getAllDataElementCodes({
+          type: 'dataValues',
+          options: {
+            GROUPING_1: {
+              dataValues: { A: { operator: '=', value: '10' } },
+            },
+            GROUPING_2: {
+              dataValues: { B: { operator: '=', value: '21' } },
+            },
+          },
+        }),
+      ).to.deep.equal(['A', 'B']));
+  });
+
+  describe('type: allOrgUnitNames', () => {
+    it('groups', () =>
+      expect(
+        groupEvents(models, EVENTS, {
+          type: 'allOrgUnitNames',
+          options: { parentCode: 'TO', type: 'district' },
+        }),
+      ).to.eventually.deep.equal({
+        Tongatapu: [
+          {
+            orgUnit: 'TO_Tongatapu',
+            orgUnitName: 'Tongatapu',
+            dataValues: [{ dataElement: 'A', value: '1' }],
           },
-          GROUPING_2: {
-            dataValues: { B: { operator: '=', value: '21' } },
+        ],
+        Niuas: [
+          {
+            orgUnit: 'TO_Niuas',
+            orgUnitName: 'Niuas',
+            dataValues: [{ dataElement: 'A', value: '2' }],
           },
-        },
-      }),
-    ).to.eventually.deep.equal({
-      GROUPING_1: [
-        {
-          dataValues: [
-            { dataElement: 'A', value: '10' },
-            { dataElement: 'B', value: '20' },
-          ],
-        },
-        {
-          dataValues: [
-            { dataElement: 'A', value: '10' },
-            { dataElement: 'B', value: '21' },
-          ],
-        },
-      ],
-      GROUPING_2: [
-        {
-          dataValues: [
-            { dataElement: 'A', value: '10' },
-            { dataElement: 'B', value: '21' },
-          ],
-        },
-      ],
+          {
+            orgUnit: 'TO_Niuas',
+            orgUnitName: 'Niuas',
+            dataValues: [{ dataElement: 'A', value: '3' }],
+          },
+        ],
+        Haapai: [],
+      }));
+
+    it('can handle non-unique org unit names', () => {
+      const EVENTS = [
+        { orgUnit: 'NZ_WELLINGTON_CITY', orgUnitName: 'Wellington', dataValues: [{ dataElement: 'A', value: '1' }] },
+        { orgUnit: 'NZ_WELLINGTON_OTHER', orgUnitName: 'Other', dataValues: [{ dataElement: 'A', value: '2' }] }, // same org unit name
+        { orgUnit: 'NZ_AUCKLAND_OTHER', orgUnitName: 'Other', dataValues: [{ dataElement: 'A', value: '3' }] },   // same org unit name
+      ]
+      return expect(
+        groupEvents(models, EVENTS, {
+          type: 'allOrgUnitNames',
+          options: { parentCode: 'NZ', type: 'district' },
+        }),
+      ).to.eventually.deep.equal({
+        Wellington: [
+          {
+            orgUnit: 'NZ_WELLINGTON_CITY',
+            orgUnitName: 'Wellington',
+            dataValues: [{ dataElement: 'A', value: '1' }],
+          },
+        ],
+        "Other (NZ_WELLINGTON_OTHER)": [
```

Comment:
I chose to put the code in the output as a disambiguator. It will of-course look bad, and isn't meant to be used like this, but instead as a way to show the data steward/developer that they should clean up their data.

The alternative of throwing an error when we come across a duplicate, I thought was too dramatic.